closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -1207,64 +1697,71 @@ private static String parseProperty(String string) {
             return string;
         }
 
-        /**
-         * Check if the brackets are balanced.
-         *
-         * @param string input string for checking
-         * @return true if the brackets are balanced
-         */
-        private static boolean areBracketsBalanced(String string) {
-            Deque<Character> deque = new LinkedList<>();
-
-            for (char ch : string.toCharArray()) {
-                switch (ch) {
-                case '{':
-                    deque.addFirst('}');
-                    break;
-                case '[':
-                    deque.addFirst(']');
-                    break;
-                case '(':
-                    deque.addFirst(')');
-                    break;
-                case '}':
-                case ']':
-                case ')':
-                    if (deque.isEmpty() || (deque.removeFirst().charValue() != ch)) {
-                        return false;
+        private static void checkProviderFormat(String providerInfo, boolean update, boolean isFIPS140_2) {
+            Pattern p = Pattern.compile(
+                "^([A-Za-z0-9\\-\\.]+)"                             // provider name
+                + "\\s*(\\s[A-Za-z0-9\\$\\{\\}\\./]+)?"             // configuration file [optional]
+                + "\\s*(\\s\\[\\s*"                                 // constraints [optional]
+                    + "([\\+\\-])?\\s*"
+                    + "([A-Za-z0-9\\{\\}\\-\\.\\=\\*\\|:,/_\\s]+)"
+                + "\\])?\\s*$");
+            Matcher m = p.matcher(providerInfo);
+            if (m.find()) {
+                if (!isFIPS140_2) {
+                    String providerName = m.group(1);
+                    if (providerName.indexOf('.') == -1) {
+                        printStackTraceAndExit("Provider must be specified using"
+                                + " the fully-qualified class name: " + providerName);
                     }
-                    break;
-                default:
-                    break;
                 }
+
+                String symbol = m.group(4);
+                if (!update && (symbol != null)) {
```

Comment:
Suggest `symbol` should be `action` and and `isNullOrBlank()` should be used (like in `setConstraints()`):
```java
                String action = m.group(4);
                if (!update && !isNullOrBlank(action)) {
```